sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java

```
@@ -0,0 +1,305 @@
+                        extractedReceiptItem.setMerchantName(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "MerchantAddress":
+                        extractedReceiptItem.setMerchantAddress(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "MerchantPhoneNumber":
+                        extractedReceiptItem.setMerchantPhoneNumber(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "Subtotal":
+                        extractedReceiptItem.setSubtotal(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "Tax":
+                        extractedReceiptItem.setTax(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "Tip":
+                        extractedReceiptItem.setTip(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "Total":
+                        extractedReceiptItem.setTotal(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "TransactionDate":
+                        extractedReceiptItem.setTransactionDate(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "TransactionTime":
+                        extractedReceiptItem.setTransactionTime(setFieldValue(fieldValue, readResults, includeTextDetails));
+                        break;
+                    case "Items":
+                        extractedReceiptItem.setReceiptItems(toReceiptItems(fieldValue.getValueArray(), readResults, includeTextDetails));
+                        break;
+                    default:
+                        break;
```

Instead of silently skipping over unknown keys, if the known set of keys are limited and new keys are not added frequently, this should throw an exception when the key is unknown. And when a new key is added, it should be accompanied by a service version update too right?

---

The leftover keys would be put in a `Map<String, FieldValue> extractedFields` to still be able to capture them and not ignore.